include/dsn/c/api_layer1.h

```diff
@@ -578,7 +578,7 @@ typedef union dsn_global_partition_id
 
 inline uint64_t dsn_gpid_to_hash(dsn_gpid gpid)
 {
-    return (((uint64_t)gpid.u.app_id) << 32) + gpid.u.partition_index;
+    return static_cast<uint64_t>(gpid.u.app_id ^ gpid.u.partition_index);
```

Comment:
@imzhenyu I also agree that the former is better. But in "replication.h", there are two hash functions for "gpid". And another is in namespace "std", which is not a very good practice for C++.